### ISSUE TITLE
M2: Ball Detection (HSV + CNN) on LRV Frames

### ISSUE THREAD

**Issue by pbranchu** [pbranchu/waterpolovids#3]
## Objective
Build the ball detection system: HSV-based candidate generation followed by CNN verification. Iterate on LRV previews (832x832) for fast development.

## Background
The water polo ball is yellow, roughly circular, and may be only a few pixels at distance. The detection pipeline is two-stage: fast deterministic candidate generation, then CNN verification to filter false positives.

LRV files (832x832 @ 25fps) are ideal for development — same content as full-res but 30x smaller and faster to process.

### 2.1 Frame Extraction & Labeling (`scripts/label_frames.py`)
- Extract ~500 diverse frames from LRV files across multiple clips
  - Sample from different clips, time offsets, lighting conditions
  - Include frames with ball visible, ball occluded, and no ball
- Export as numbered PNGs to `data/labeling/frames/`
- Labeling options:
  - Simple script: display frame, click to mark ball center + radius
  - Or export to Roboflow/CVAT for bbox annotation, export YOLO format
- Output: annotations file mapping frame → ball bbox (or "no ball")

### 2.2 Stage A — HSV Candidate Generation (`src/wpv/tracking/detector.py`)
Per frame:
1. **Convert BGR → HSV**
2. **Multi-band yellow threshold** — multiple HSV ranges to handle:
   - Bright daylight (high saturation, high value)
   - Indoor/cloudy (lower saturation)
   - Wet ball (shifted hue, lower saturation)
   - Suggested starting ranges:
     ```python
     HSV_BANDS = [
         ((18, 100, 100), (35, 255, 255)),  # standard yellow
         ((15, 60, 120), (40, 255, 255)),    # washed-out yellow
     ]
     ```
3. **Morphological cleanup** — open then close (remove noise, fill gaps)
4. **Connected components** — extract blobs
5. **Filter candidates** by:
   - Pixel area bounds (configurable, scale with resolution)
   - Circularity (perimeter² / area, threshold ~0.6+)
   - Mean saturation/value within blob
   - Pool ROI constraint (optional mask to exclude non-pool areas)
6. **Output**: list of `Candidate(bbox, centroid, area, circularity, mean_hsv, confidence_features)`

### 2.3 Stage B — CNN Verification
- Crop each candidate region (with padding) from the frame
- Resize to fixed input size (e.g. 64x64 or 96x96)
- Run through lightweight binary classifier:
  - **Option A**: MobileNetV3-Small (pretrained, fine-tune last layers)
  - **Option B**: YOLOv8-nano trained on ball crops
  - Start with Option A — simpler, faster to train
- Output: confidence score per candidate
- Only candidates above threshold (e.g. 0.7) pass to tracker
- Training:
  - Positive: labeled ball crops from 2.1
  - Negative: HSV false positives (splash reflections, lane markers, yellow caps)
  - Data augmentation: rotation, brightness, blur, scale
  - Target: >95% precision at >85% recall

### 2.4 Detection Pipeline Integration
- `BallDetector` class combining Stage A + Stage B
- Process a full video (LRV) frame-by-frame, output detections JSON
- Visualization: draw detected candidates (green=verified, red=rejected) on frames
- Export annotated video for visual inspection

### 2.5 CLI Commands
```bash
wpv detect-ball <video_path> --output detections.json     # Run detector
wpv detect-ball <video_path> --visualize --output vis.mp4  # Annotated video
wpv label-frames <video_path> --count 100 --output frames/ # Extract frames for labeling
```

## Validation
- [ ] HSV stage produces candidates on most frames where ball is visible
- [ ] CNN correctly classifies >95% of ball crops as positive
- [ ] CNN rejects >90% of false positives (splashes, caps, reflections)
- [ ] Visual overlay video shows green boxes tracking the ball across frames
- [ ] Detection runs at >10fps on LRV (832x832) on local GPU
- [ ] Unit tests for HSV thresholding and candidate filtering
